Assets/MRTK.ThirdParty/MRTK-Quest/Editor/MRTKQuestConfigurationChecker.cs

```diff
@@ -0,0 +1,204 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace prvncher.MixedReality.Toolkit.OculusQuestInput
+{
+    /// <summary>
+    /// Class that checks if the Oculus Integration Assets are present and configures the project if they are.
+    /// </summary>
+    [InitializeOnLoad]
+    static class OculusConfigurationChecker
+    {
+        // The presence of the OculusProjectConfig.asset is used to determine if the Oculus Integration Assets are in the project.
+        private const string OculusIntegrationProjectConfig = "OculusProjectConfig.asset";
+        private static readonly string[] Definitions = { "OCULUSINTEGRATION_PRESENT" };
+
+        static OculusConfigurationChecker()
+        {
+            // Check if Oculus Integration Package is in the project
+            ReconcileOculusIntegrationDefine();
+        }
+
+        //seems to work fine as is
+        private static bool ReconcileOculusIntegrationDefine()
+        {
+            FileInfo[] files = FindFilesInAssets(OculusIntegrationProjectConfig);
+
+            if (files.Length > 0)
+            {
+                AppendScriptingDefinitions(BuildTargetGroup.Android, Definitions);
+                AppendScriptingDefinitions(BuildTargetGroup.Standalone, Definitions);
+                return true;
+            }
+            else
+            {
+                RemoveScriptingDefinitions(BuildTargetGroup.Android, Definitions);
+                RemoveScriptingDefinitions(BuildTargetGroup.Standalone, Definitions);
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Configures the project profiles to support handtracking
+        /// </summary>
+        [MenuItem("Mixed Reality Toolkit/Utilities/Oculus/Configure Oculus Profiles for Handtracking")]
+        static void ConfigureProfilesForHandtracking()
+        {
+            FileInfo[] files = FindFilesInAssets(OculusIntegrationProjectConfig);
+
+            // Make this set the application to controllers and hands on first setup
+#if OCULUSINTEGRATION_PRESENT
```

Comment:
Nit: Move this before FindFilesInAssets because it's not used if OCULUSINTEGRATION_PRESENT isn't set